docs/advanced-setup.md

```diff
@@ -32,6 +32,9 @@ wisely if `volumeBindingMode` is `Immediate`.
 You can use it to set `device-class` that the StorageClass will use.
 The `device-class` is described in the [LVMd](lvmd.md) document.
 
+`reclaimPolicy` can be either `Delete` or `Retain`.
+If you delete a PVC whose corresponding PV has `Retain` reclaim policy, the corresponding `LogicalVolume` resource and the LVM logical volume are *NOT* deleted. If you delete this `LogicalVolume` resource after deleting the PVC, the related LVM logical volume is also deleted.
```

Comment:
I think the correct way to guide users would be to delete the PersistentVolume which will trigger a delete of the underlying LogicalVolume as well right?

Reply:
I'm afraid not. Deleting a PV doesn't trigger a delete of the underlying LogicalVolume, as shown below:

```
$ cd topolvm/test/e2e
$ make start-lvmd
$ make create-cluster
$ cat topolvm-provisioner-retain.yaml 
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: topolvm-provisioner-retain
provisioner: topolvm.io
reclaimPolicy: Retain
volumeBindingMode: WaitForFirstConsumer
$ kubectl apply -f topolvm-provisioner-retain.yaml
$ cat test-pvc.yaml
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: my-pvc
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 1Gi
  storageClassName: topolvm-provisioner-retain
$ kubectl apply -f test-pvc.yaml
persistentvolumeclaim/my-pvc created
$ cat test-pod.yaml
apiVersion: v1
kind: Pod
metadata:
  name: my-pod
spec:
  containers:
  - name: pause
    image: registry.k8s.io/pause
    volumeMounts:
    - mountPath: /data
      name: volume
  volumes:
  - name: volume
    persistentVolumeClaim:
      claimName: my-pvc
$ kubectl apply -f test-pod.yaml
pod/my-pod created
$ sudo lvs
  LV                                   VG           Attr       LSize   Pool Origin Data%  Meta%  Move Log Cpy%Sync Convert
  ddee77d7-7d78-4790-8bd1-2537a871785f node1-thick1 -wi-ao----   1.00g                                                    
  pool0                                node1-thin1  twi-a-tz--   4.00g             0.00   11.23                           
  pool0                                node2-thin1  twi-a-tz--   4.00g             0.00   11.23                           
  pool0                                node3-thin1  twi-a-tz--   4.00g             0.00   11.23                           
  ubuntu-lv                            ubuntu-vg    -wi-ao---- 509.99g                                                    
$ kubectl delete pod my-pod
pod "my-pod" deleted
$ kubectl delete pvc my-pvc
persistentvolumeclaim "my-pvc" deleted
$ kubectl get pv
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS     CLAIM            STORAGECLASS                 REASON   AGE
pvc-09635214-facb-4ecc-9275-788066c12381   1Gi        RWO            Retain           Released   default/my-pvc   topolvm-provisioner-retain            25s
$ kubectl delete pv pvc-09635214-facb-4ecc-9275-788066c12381
persistentvolume "pvc-09635214-facb-4ecc-9275-788066c12381" deleted
$ sudo lvs
  LV                                   VG           Attr       LSize   Pool Origin Data%  Meta%  Move Log Cpy%Sync Convert
  ddee77d7-7d78-4790-8bd1-2537a871785f node1-thick1 -wi-a-----   1.00g    ## <------ Still there.
  pool0                                node1-thin1  twi-a-tz--   4.00g             0.00   11.23                           
  pool0                                node2-thin1  twi-a-tz--   4.00g             0.00   11.23                           
  pool0                                node3-thin1  twi-a-tz--   4.00g             0.00   11.23                           
  ubuntu-lv                            ubuntu-vg    -wi-ao---- 509.99g                                                    
```

See also #710.

Note that this seems to be the correct behavior of the `Retain` reclaim policy. [The k8s document](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#retain) says:

> Delete the PersistentVolume. The associated storage asset in external infrastructure still exists after the PV is deleted.